### PR TITLE
Fixup pin definitions for UART + Analog

### PR DESCRIPTION
**Pull request by tyeth**
Removes duplicate pin and fixes order in QT Py ESP32 Pico.
Specify all PWM pins as servo+pwm compatible.
Restrict Feather ESP32 v1 A2-A4 input only

### CODE REVIEW

**Comment by tyeth** [comment]
@brentru double checked, think I got them all

File: boards/nano-33-iot/definition.json [hunk 1]
```diff
@@ -9,6 +9,18 @@
     "installMethod":"library",
     "components":{
        "digitalPins":[
+          {
+             "name":"TX",
```
Comment (brentru) [inline boards/nano-33-iot/definition.json:13]: @tyeth  is there any `D` prefixed name for this, or is it just TX within the board definition?

File: boards/qtpy-esp32s2/definition.json [hunk 2]
```diff
@@ -104,26 +118,44 @@
...
                 "displayName":"SCL",
-                "direction":"INPUT",
-                "dataType":"int16"
+                "dataType":"int16",
+                "hasPWM":true,
+                "hasServo":true
+            },
+            {
+                "name":"A5",
+                "displayName":"A5 (UART TX)",
```
Comment (brentru) [inline boards/qtpy-esp32s2/definition.json:148]: why is this not `uartTx`?

**Review by brentru — CHANGES_REQUESTED** [review]
I have some questions to resolve, then we can merge in. Excellent job on adding missing links to documentation while you were in the .json files!

File: boards/nano-33-iot/definition.json (hunk 1 above)
Comment (tyeth) [reply boards/nano-33-iot/definition.json:13]: Looks like not:
https://github.com/espressif/arduino-esp32/blob/master/variants/nano32/pins_arduino.h
https://docs.arduino.cc/static/d77fe4114922e80804b6ebfd2e586316/ABX00027-pinout.png

Comment (tyeth) [reply boards/nano-33-iot/definition.json:13]: Looks like not, this is the pinout:
https://docs.arduino.cc/static/d77fe4114922e80804b6ebfd2e586316/ABX00027-pinout.png
This is the definition:
https://github.com/arduino/ArduinoCore-samd/blob/master/variants/nano_33_iot/variant.cpp#L35-L36

Technically we could support the second naming system of PBxx/PAxx for all pins, but recommend against that.

File: boards/qtpy-esp32s2/definition.json (hunk 2 above)
Comment (tyeth) [reply boards/qtpy-esp32s2/definition.json:148]: Not part of the analog pins schema, should it be?

File: boards/qtpy-esp32s2/definition.json [hunk 3]
```diff
@@ -104,26 +118,44 @@
...
+                "dataType":"int16",
+                "hasPWM":true,
+                "hasServo":true
+            },
+            {
+                "name":"A16",
+                "displayName":"A16 (UART RX)",
```
[the remark this answers is not in the document]
Comment (tyeth) [reply boards/qtpy-esp32s2/definition.json:155]: Not part of the analog pins schema, should it be?

File: boards/qtpy-esp32/definition.json [hunk 4]
```diff
@@ -158,6 +151,20 @@
                 "dataType":"int16",
                 "hasPWM":true,
                 "hasServo":true
+            },
+            {
+                "name":"A32",
+                "displayName":"A32 (UART TX)",
+                "dataType":"int16",
+                "hasPWM":true,
+                "hasServo":true
```
[the remark this answers is not in the document]
Comment (tyeth) [reply boards/qtpy-esp32/definition.json:160]: Not part of the analog pins schema, should it be?

**Review by brentru — APPROVED** [review]

**Pull request merged**